TombEngine/Renderer/Renderer11Draw2D.cpp

```diff
@@ -373,74 +375,147 @@ namespace TEN::Renderer
 		m_context->RSSetViewports(1, &m_viewport);
 		ResetScissor();
 
-		DrawFullScreenQuad(texture, Vector3(fade, fade, fade), true);
+		DrawFullScreenQuad(texture, Vector3(fade), true);
 	}
 
-	void Renderer11::DrawFullScreenQuad(ID3D11ShaderResourceView* texture, DirectX::SimpleMath::Vector3 color, bool fit)
+	void Renderer11::DrawSpriteIn2DSpace(GAME_OBJECT_ID spriteID, unsigned int spriteIndex, const Vector2& pos2D, short orient2D,
+										 const Vector4& color, const Vector2& size)
 	{
-		Vector2 uvStart = { 0.0f, 0.0f };
-		Vector2 uvEnd   = { 1.0f, 1.0f };
+		constexpr auto VERTEX_COUNT	 = 4;
+		constexpr auto UV_CONSTRAINT = std::pair<Vector2, Vector2>(Vector2(0.0f), Vector2(1.0f));
+
+		// Calculate vertex base.
+		auto halfSize = size / 2;
+		auto vertexPoints = std::array<Vector2, VERTEX_COUNT>
+		{
+			halfSize,
+			Vector2(-halfSize.x, halfSize.y),
+			-halfSize,
+			Vector2(halfSize.x, -halfSize.y)
+		};
+
+		// Transform vertices.
+		auto rotMatrix = Matrix::CreateRotationZ(TO_RAD(orient2D + ANGLE(180.0f))); // NOTE: +Y is down.
+		for (auto& vertexPoint : vertexPoints)
+		{
+			// Rotate.
+			vertexPoint = Vector2::Transform(vertexPoint, rotMatrix);
+
+			// Adjust for aspect ratio and convert to NDC.
+			vertexPoint = TEN::Utils::GetAspectCorrect2DPosition(vertexPoint);
+			vertexPoint += pos2D;
+			vertexPoint = TEN::Utils::Convert2DPositionToNDC(vertexPoint);
+		}
+
+		g_Renderer.PrintDebugMessage("%.3f, %.3f", vertexPoints[0].x, vertexPoints[0].y);
+
+		// Define renderer vertices.
+		auto vertices = std::array<RendererVertex, VERTEX_COUNT>{};
+
+		// Vertex 0
+		vertices[0].Position = Vector3(vertexPoints[0]);
```

Comment:
this might be a weird idea but how about having all of the vertex 0-3 setup in a for loop? 

For example if the number is odd, then process for first x and second y and if it is even then process for second UV constraint